packages/model-generator/src/get_backend_output_with_error_handling.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  BackendOutputClient,
+  BackendOutputClientError,
+  BackendOutputClientErrorType,
+  DeployedBackendIdentifier,
+} from '@aws-amplify/deployed-backend-client';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+
+/**
+ * Common Error handling for BackendOutputClient.getOutput() in model-generator package.
+ */
+export const getBackendOutputWithErrorHandling = (
+  backendOutputClient: BackendOutputClient,
+  backendIdentifier: DeployedBackendIdentifier
+) => {
```

Comment:
This could be turned into OOP .
It looks like candidate for adapter pattern, i.e. `new BackendOutputClientWithErrorHandlingAdapter(backendOutputClient)` .

Reply:
What's wrong with a function? As Ed mentioned [here](https://github.com/aws-amplify/amplify-backend/pull/1734#discussion_r1676378018), they are both valid options?

Reply:
They are but OOP and DI might be better long term.

If we do OOP:
1. all tests with name `throws an error if deployment is currently in progress` could be reduced to single test for `BackendOutputClientWithErrorHandlingAdapter`
2. Components requiring error handler behavior would rely on typing ctor parameters and compiler.

This isn't a big deal in this PR, but if this pattern spreads it's going to be harder to refactor.

to be clear this comment is a suggestion, not a hard ask.

Reply:
We can always move the tests to the new function instead of the call site. Since this is pretty internal to this package and I don't expect the logic here to grow, we can keep this for now and reevaluate later.